File: JavaScript/Array&String/605. Can Place Flowers.js
/* 605. Can Place Flowers
Difficulty: Easy
Submission Link: https://leetcode.com/problems/can-place-flowers/submissions/1017427192
 */

var canPlaceFlowers = function(flowerbed, n) {
    let count = 0;
    for (let i = 0; i < flowerbed.length; i++) {
        if (flowerbed[i] === 0) {
            if ((i === 0 || flowerbed[i - 1] === 0) && (i === flowerbed.length - 1 || flowerbed[i + 1] === 0)) {
                flowerbed[i] = 1;
                count++;
            }
        }
    }
    return count >= n;
};

File: Python/Array&String/605. Can Place Flowers.py
"""
605. Can Place Flowers
Difficulty: Easy
Submission Link:https://leetcode.com/problems/can-place-flowers/submissions/1017424240"""

class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        count = 0
        i = 0
        
        while i < len(flowerbed):
            if flowerbed[i] == 0:
                if i == 0 or flowerbed[i - 1] == 0:
                    if i == len(flowerbed) - 1 or flowerbed[i + 1] == 0:
                        flowerbed[i] = 1
                        count += 1
            i += 1
            
        return count >= n